Bottom: e401291a2b8aa24d37ca7891d028425ddd6777fc
Top:    7b9643971d84886323de7526e173c5d0af5c2163
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-24 00:11:43 +0100

polygon1.c: Fixup warnings due to assigned but unused variables

Adds a nasty cludge using #ifndef NDEBUG around a variable assignment
which is only used in a later assert() statement. (NB: assert()
evaluates to nothing if NDEBUG is defined).


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 70d553f..ffb1542 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1980,7 +1980,7 @@ M_POLYAREA_update_primary (jmp_buf * e, POLYAREA ** pieces,
   POLYAREA *anext;
   PLINE *curc, *next, *prev;
   BoxType box;
-  int inv_inside = 0;
+  /* int inv_inside = 0; */
   int del_inside = 0;
   int del_outside = 0;
   int finished;
@@ -1998,7 +1998,7 @@ M_POLYAREA_update_primary (jmp_buf * e, POLYAREA ** pieces,
       del_inside = 1;
       break;
     case PBO_XOR:		/* NOT IMPLEMENTED OR USED */
-      inv_inside = 1;
+      /* inv_inside = 1; */
       assert (0);
       break;
     }
@@ -2649,7 +2649,9 @@ void
 poly_InvContour (PLINE * c)
 {
   VNODE *cur, *next;
+#ifndef NDEBUG
   int r;
+#endif
 
   assert (c != NULL);
   cur = &c->head;
@@ -2664,8 +2666,13 @@ poly_InvContour (PLINE * c)
   c->Flags.orient ^= 1;
   if (c->tree)
     {
-      r = r_search (c->tree, NULL, NULL, flip_cb, NULL);
+#ifndef NDEBUG
+      r =
+#endif
+        r_search (c->tree, NULL, NULL, flip_cb, NULL);
+#ifndef NDEBUG
       assert (r == c->Count);
+#endif
     }
 }
